src/algorithm_tiled.erl

```diff
@@ -32,19 +32,30 @@ run(Steps, Env, Config) ->
 step(MaxT, MaxT, Env, _Pool, _Config) ->
     Env;
 step(T, MaxT, Env, Pool, Config) ->
+  %io:format("Step ~p~n", [T]),
     NColours = 2,
     NParts = Config#config.tiles_per_colour,
-    Partitioned = agents:partition(Env, NColours, NParts, Config),
+    Partitioned = algorithm:partition(Env, NColours, NParts, Config),
     ProcessColour =
         fun(Colour, E) ->
-                SendToWork = fun(Agents) ->
-                                     send_to_work(Pool, Agents, E, Config)
+                SendToWork = fun(Range) ->
+                                     Tile=agents:get_tile(Range, E, Config),
+                                     send_to_work(Pool, Tile, E, Config)
                              end,
                 lists:map(SendToWork, Colour),
-                wait(Colour),
-                E
+                NewEnvs = collect_results(Colour),
+%%           NA = [{agent,Pos,{Dir,State}} || {agent,Pos,{Dir,State}} <- lists:flatten([Agents || {_, {env, Agents, _}} <- NewEnvs]), Dir=/=empty],
+%%           io:format("Step ~p After processing one color: ~p~n", [T, NA]),
+                NE = agents:update_tiles(NewEnvs, E, Config),
+%%           {env, Agents, _} = NE,
+%%           Test = [{agent,Pos,{Dir,State}} || {agent,Pos,{Dir,State}}  <- Agents, Dir=/=empty],
+%%           io:format("Step ~p Agents after update: ~p~n", [T, Test]),
+          NE
         end,
     NewEnv = lists:foldl(ProcessColour, Env, Partitioned),
+%%   {env, Agents, _} = NewEnv,
+%%   Test = [{agent,Pos,{Dir,State}} || {agent,Pos,{Dir,State}}  <- Agents, Dir=/=empty],
```

Comment:
Line 57 is too long: %%   Test = [{agent,Pos,{Dir,State}} || {agent,Pos,{Dir,State}}  <- Agents, Dir=/=empty],.